org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/internal/core/variables/IJavaValue.java

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+* Copyright (c) 2017 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.internal.core.variables;
+
+/**
+ * An object, primitive data type, or array, on a java custom object.
+ */
+public interface IJavaValue {
```

Comment:
This interface is an abstract of a variable. So suggest IVariable. It should have a name, a type and a value.